### PR TITLE
Convert react-clipboard to full ES6 to remove React v15.5.0 deprecation warnings

### PR DESCRIPTION
**Pull request by richtr**
This update removes deprecation warnings introduced in React v15.5.0
for React.PropTypes and React.createClass that were used in this
package.

(see: https://facebook.github.io/react/blog/2017/04/07/react-v15.5.0.html#new-deprecation-warnings)

### CODE REVIEW

**Comment by richtr** [comment]
React Event Handling was also upgraded according to https://facebook.github.io/react/docs/handling-events.html.

**Comment by richtr** [comment]
The only thing I'm not sure about is whether to add `"prop-types": "^15.5.0",` to `peerDependencies` or `dependencies` in `package.json`.

If the user does not use `prop-types` elsewhere this could then fail, right @WickyNilliams?

**Comment by richtr** [comment]
Following on from my previous comment, `package.json` should now be correct after 51f0abb745f20dac6c01f85bd368c732ff1051a4.

**Comment by WickyNilliams** [comment]
Hey Rich! Cheers for this. Looks good on the surface. I'll test over the weekend and merge as soon as i can. I should really write some tests for this :)

**Comment by WickyNilliams** [comment]
What do you mean by this:

> React Event Handling was also upgraded according to facebook.github.io/react/docs/handling-events.html.

And yes, it's best to have prop-types as a regular dependency

File: src/index.js
Comment (WickyNilliams) [inline src/index.js]: could you move these up so they're static properties on the class? e.g.

```js
class Clipboard extends React.Component {
  static propTypes = {};
}
```

**Review by WickyNilliams — CHANGES_REQUESTED** [review]

**Comment by richtr** [comment]
I have fixed up all review feedback now.

Regarding the following:

>What do you mean by this:
>
>> React Event Handling was also upgraded according to facebook.github.io/react/docs/handling-events.html.

I just mean that we use this pattern now for events on the `Clipboard` class: `handleKeyDown = e => {` as documented at that URL.

File: src/index.js
Comment (WickyNilliams) [inline src/index.js]: This line should be:

```js
static defaultProps = {
```

File: package.json
Comment (WickyNilliams) [inline package.json]: Why stage-1 instead of the specific transform that you originally had? This approach might potentially cause a breakage when the feature moves beyond stage-1, right?

File: package.json
Comment (WickyNilliams) [inline package.json:33]: Some work is going to have to be done before this PR can be merged. With the addition of a dependency, a bundler is required so that `prop-types` gets included in the build output (currently only babel is used for building, which doesn't do any bundling itself).

I'm happy to do this, but it may be a few days before I get around to it.

**Review by WickyNilliams — CHANGES_REQUESTED** [review]

File: package.json
Comment (WickyNilliams) [reply package.json:33]: Actually, I take all this back :) It's not required.

When a user installs react-clipboard, dependencies will be pulled down, and when they bundle, dependencies will be included alongside their application code. All good.

File: src/index.js
Comment (richtr) [reply src/index.js]: Oops that's a typo!

Fixed in 18ef11e724d8c24e71b7580841b5d8271e0d4dc2.

File: package.json
Comment (richtr) [reply package.json]: I re-instated the `transform-class-properties` plugin approach as requested.

I also replaced `babel-preset-es2015` with `babel-preset-env` to remove this warning on `npm install`:

> npm WARN deprecated babel-preset-es2015@6.24.1: We're super 😸  excited that you're trying to use ES2015 syntax, but instead of making more yearly presets 😭 , Babel now has a better preset that we recommend you use instead: npm install babel-preset-env --save-dev. preset-env without options will compile ES2015+ down to ES5 just like using all the presets together and thus is more future proof. It also allows you to target specific browsers so that Babel can do less work and you can ship native ES2015+ to user 😎 ! We are also in the process of releasing v7, so please give http://babeljs.io/blog/2017/09/12/planning-for-7.0 a read and help test it out in beta! Thanks so much for using Babel 🙏, please give us a follow on Twitter @babeljs for news on Babel, join slack.babeljs.io for discussion/development and help support the project at opencollective.com/babel

All done in b5b47a5d1ae5f7031fef821e16844ad76745c9ea

File: package.json
Comment (richtr) [reply package.json:33]: 👍

**Comment by richtr** [comment]
Thanks for reviewing this @WickyNilliams and some good questions there. Everything should be fixed up and good to go but let me know if there is anything else that needs discussing or changing.

All the best! Great working with you again :)

**Review by WickyNilliams — APPROVED** [review]

**Comment by WickyNilliams** [comment]
Published to npm as `v1.3.2`.

Thanks again Rich!

**Pull request merged**
